packages/common/src/compiler/exploreCompiler.ts

```diff
@@ -83,22 +83,14 @@ export type UncompiledExplore = {
 const getReferencedTable = (
     refTable: string,
     tables: Record<string, Table>,
-    currentTable: string,
```

Comment:
@rephus I rolled back most of your fix from https://github.com/lightdash/lightdash/pull/9842 since the tables arg already has the joined tables under the correct alias. 

eg:
```
tables {
   orders: ...
   customers: ...
   customers_alias: ...
}
```

Reply:
>  Consideration when referencing joined models. If the model with the metric is joined to other models it will throw an error (see image bellow) unless that model also has the referred model.

mm, is it this the same use case though ? 
- I have a model Customers (A) 
- I have another model orders (B) joins a model (A) with an alias, and uses the alias from model A in a metric. 

I'm not using this `foo` reference on another table like payments or similar. 

> since the tables arg already has the joined tables under the correct alias.

In this case, I'd expect to see `foo` inside the `tables` when `currentTable`=== orders 

```
tables {
  payments: {},
  orders: {},
  customers: {},
  foo: {}, // same content as customers
 }
  ```

Reply:
> mm, is it this the same use case though ?

Yes, let's use my example and replace the names with your code. 

Example:
model A (**orders**) joins model B (**foo**) and has a metric (**test_customers_foo**) referencing model B (**foo**).
model C (**payments**) joins model A(**orders**). This will result in an error unless we explicitly join model B(**foo**) to model C(**payments**).

> I'm not using this foo reference on another table like payments or similar.

When we compile `payments` explore we compile `orders` table which has the reference to `foo`. 

> In this case, I'd expect to see foo inside the tables when currentTable=== orders

`tables` arg has all the tables for the explore being compiled. 

`orders` is compiled without errors because `foo` is in `tables`.
`payments` throws an error because only `payments`, `orders`, `customers` are part of the `tables`.


Your example with payments only seemed to work because it was already joining `customers`. If `payments` doesn't join `customers` it would result in an error. The other assumption is that the join SQL would be the same between `foo` from orders join and `customers` from payments join